### PR TITLE
Use Travis CI to auto update `gh-pages` (fix #296)

### LINKED ISSUE #296: Please autodeploy the built spec
CC @bterlson @domenic

### PR DESCRIPTION
**Pull request by alrra**
* Move [`figure-1.svg`](https://github.com/tc39/ecma262/blob/70562e82b7a8880375fec44f35facb174ebebed9/figure-1.svg) and the images from the `gh-pages` branch (namely: [`favicon.ico`](https://github.com/tc39/ecma262/blob/d226c076305cddedd5eb4b6c4dccd61c9eb0049c/favicon.ico), [`figure-1.png`](https://github.com/tc39/ecma262/blob/d226c076305cddedd5eb4b6c4dccd61c9eb0049c/figure-1.png) & [`figure-2.png`](https://github.com/tc39/ecma262/blob/d226c076305cddedd5eb4b6c4dccd61c9eb0049c/figure-2.png) ) into `img/`, and update the build step to copy them into `out/`.

* Add a `.travis.yml` file with configurations to make Travis CI automatically update the content from the `gh-pages` branch with the content from `out/` whenever a change is made in the `master` branch and the build is successful.

--

:warning: __edited__

@bterlson

First off, read [this](https://github.com/alrra/travis-scripts/blob/master/doc/github-deploy-keys.md) to understand how the deploy keys will work.

Now, in order for the code from this pull request to work, you will need to update some things:

* Add [your own encrypted private key](https://github.com/alrra/travis-scripts/blob/master/doc/github-deploy-keys.md#25-encrypt-the-file-containing-the-private-key) (replace [`scripts/travis/github_deploy_key.enc`](https://github.com/alrra/ecma262/blob/fix-296/scripts/travis/github_deploy_key.enc))
* Replace [`$encrypted_XXXXXXXXXXXX_key` and `$encrypted_XXXXXXXXXXXX_iv`](https://github.com/alrra/ecma262/blob/fix-296/scripts/travis/auto-deploy.sh#L10-L11) with the [appropriate values](https://github.com/alrra/travis-scripts/blob/master/doc/github-deploy-keys.md#25-encrypt-the-file-containing-the-private-key)
* Provide [the environment variables `GH_USER_NAME` and `GH_USER_EMAIL`](https://github.com/alrra/travis-scripts/blob/master/doc/github-deploy-keys.md#3-add-the-environment-variables-gh_user_name-and-gh_user_email) by replacing [`secure: "<secure_key_value>"`](https://github.com/alrra/ecma262/blob/fix-296/.travis.yml#L6)

--

@bterlson Let me know if I can help you with something, or if I need to change anything.

### CODE REVIEW

**Comment by bterlson** [comment]
@alrra I will take a look at this tomorrow! But let me say, thanks so much for making this easy for me. It's pretty great you've given me a checklist (with screenshots!).

**Comment by annevk** [comment]
❤️

**Comment by mathiasbynens** [comment]
In step 2.iii, if you append `--add env.global` to the command, it will automatically add the key to your `.travis.yml` file.

**Comment by alrra** [comment]
>  --add env.global to the command, it will automatically add the key to your .travis.yml file.

The reason why I didn't suggested it was because last night when I tested to see that everything works, `--add env.global` screwed up the `.travis.yml` file.

**Comment by bterlson** [comment]
This looks mostly good. I would prefer it if the images didn't have a different path in relative to spec.html in the different branches though. I like the img subfolder though, maybe we can just copy the entire folder to gh-pages and update spec.html to point to this location?

**Comment by michaelficarra** [comment]
It would make me uncomfortable that I was giving Travis and this npm module access to write to *all* public repos I own. In [my own proposal](https://github.com/michaelficarra/Function-prototype-toString-revision), I use a deploy key that only has access to that one repo.

**Comment by alrra** [comment]
> maybe we can just copy the entire folder to gh-pages and update spec.html to point to this location?

@bterlson Sure, I will update my pull request.

> deploy key that only has access to that one repo.

@michaelficarra Thanks for your comment! For some reason I thought there were similar in regards to access, my mistake. I'll update the Travis script.

File: package.json [hunk 1]
```diff
@@ -15,5 +15,8 @@
   "homepage": "https://tc39.github.io/ecma262/",
   "dependencies": {
     "ecmarkup": "^3.0.0"
+  },
+  "devDependencies": {
+    "@alrra/travis-scripts": "^1.2.2"
```
Comment (michaelficarra) [inline package.json]: Additionally, I would lock down the version of this dependency (and probably npm shrinkwrap) to protect against malicious updates.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+                             --source-branch "master"
+
+env:
+  global:
+    - secure: "<secure_key>"
+
+language: node_js
+
+node_js:
+  - '5'
```
Comment (michaelficarra) [inline .travis.yml]: Can we not mix single/double quotes? This is the only single-quoted value in this file.

File: package.json (hunk 1 above)
Comment (ljharb) [inline package.json]: locking down dep versions doesn't give you any protection without shrinkwrap, and fails to communicate the intent to follow semver for updates when updating - it'd be best to use a `^` range, and then if you're concerned about updates, `npm shrinkwrap --dev` on top of that.

**Comment by Ms2ger** [comment]
What's this blocked on?

**Comment by michaelficarra** [comment]
Probably this: https://github.com/tc39/ecma262/pull/309#issuecomment-174166968

**Comment by bterlson** [comment]
Everything now auto-deploys!

**Comment by UltCombo** [comment]
Yayy! :clap:

**Pull request closed without merging**